test/integration/targets/vmware_guest_serial_port/tasks/main.yml

```diff
@@ -0,0 +1,100 @@
+# Test code for the vmware_guest_serial_port Operations.
+# Copyright: (c) 2019, Anusha Hegde <anushah@vmware.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- import_role:
+    name: prepare_vmware_tests
+
+- name: Create VM
+  vmware_guest:
+    validate_certs: False
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    name: test_vm1
+    guest_id: centos64Guest
+    datacenter: "{{ dc1 }}"
+    folder: "{{ f0 }}"
+    hardware:
+      num_cpus: 1
+      memory_mb: 512
+    disk:
+      - size: 1gb
+        type: thin
+        autoselect_datastore: True
+    state: present
+  register: create_vm_for_test
+
+- name: assert that changes were made
+  assert:
+    that:
+        - create_vm_for_test is changed
+
+- name: Create multiple serial ports with Backing type - network, pipe, device and file
```

Comment:
This step fails with the following error:

```
TASK [vmware_guest_serial_port : Create multiple serial ports with Backing type - network, pipe, device and file] ****************************************************************************
task path: /home/goneri/.ansible/test/tmp/vmware_guest_serial_port-_6233y53-ÅÑŚÌβŁÈ/test/integration/targets/vmware_guest_serial_port/tasks/main.yml:36
<testhost> ESTABLISH LOCAL CONNECTION FOR USER: goneri
<testhost> EXEC /bin/sh -c 'echo ~goneri && sleep 0'
<testhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555 `" && echo ansible-tmp-1571348205.6692333-209565769524555="` echo /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555 `" ) && sleep 0'                                                                                             
Using module file /home/goneri/git_repos/ansible_projects/b/lib/ansible/modules/cloud/vmware/vmware_guest_serial_port.py
<testhost> PUT /home/goneri/.ansible/tmp/ansible-local-27978cc47up40/tmpy1djlbkh TO /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/AnsiballZ_vmware_guest_serial_port.py                                                                                                                                                                                          
<testhost> EXEC /bin/sh -c 'chmod u+x /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/ /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/AnsiballZ_vmware_guest_serial_port.py && sleep 0'                                                                                                                                                  
<testhost> EXEC /bin/sh -c 'ESXI1_HOSTNAME=esxi3.test ESXI1_PASSWORD=root ESXI1_USERNAME=root ESXI2_HOSTNAME=test4.test ESXI2_PASSWORD=root ESXI2_USERNAME=root VCENTER_HOSTNAME=vcenter.test VCENTER_PASSWORD='"'"'!234AaAa56'"'"' VCENTER_USERNAME=administrator@vsphere.local VMWARE_VALIDATE_CERTS=false /tmp/python-wq6lw5ae-ansible/python /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/AnsiballZ_vmware_guest_serial_port.py && sleep 0'                                                                                                               
<testhost> EXEC /bin/sh -c 'rm -f -r /home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/ > /dev/null 2>&1 && sleep 0'
fatal: [testhost]: FAILED! => {
    "changed": false,
    "module_stderr": "pyVmomi.VmomiSupport.InvalidDatastorePath: (vim.fault.InvalidDatastorePath) {\n   dynamicType = <unset>,\n   dynamicProperty = (vmodl.DynamicProperty) [],\n   msg = \"Invalid datastore path 'newfile'.\",\n   faultCause = <unset>,\n   faultMessage = (vmodl.LocalizableMessage) [],\n   datastore = <unset>,\n   name = <unset>,\n   datastorePath = 'newfile'\n}\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/AnsiballZ_vmware_guest_serial_port.py\", line 116, in <module>\n    _ansiballz_main()\n  File \"/home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/AnsiballZ_vmware_guest_serial_port.py\", line 108, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/home/goneri/.ansible/tmp/ansible-tmp-1571348205.6692333-209565769524555/AnsiballZ_vmware_guest_serial_port.py\", line 54, in invoke_module\n    runpy.run_module(mod_name='ansible.modules.cloud.vmware.vmware_guest_serial_port', init_globals=None, run_name='__main__', alter_sys=False)\n  File \"/usr/lib64/python3.7/runpy.py\", line 208, in run_module\n    return _run_code(code, {}, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.7/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_vmware_guest_serial_port_payload_wgxarmo1/ansible_vmware_guest_serial_port_payload.zip/ansible/modules/cloud/vmware/vmware_guest_serial_port.py\", line 454, in <module>\n  File \"/tmp/ansible_vmware_guest_serial_port_payload_wgxarmo1/ansible_vmware_guest_serial_port_payload.zip/ansible/modules/cloud/vmware/vmware_guest_serial_port.py\", line 438, in main\n  File \"/tmp/ansible_vmware_guest_serial_port_payload_wgxarmo1/ansible_vmware_guest_serial_port_payload.zip/ansible/modules/cloud/vmware/vmware_guest_serial_port.py\", line 242, in reconfigure_vm_serial_port\n  File \"/tmp/ansible_vmware_guest_serial_port_payload_wgxarmo1/ansible_vmware_guest_serial_port_payload.zip/ansible/module_utils/vmware.py\", line 82, in wait_for_task\n  File \"<string>\", line 3, in raise_from\nansible.module_utils.vmware.TaskError: (\"Invalid datastore path 'newfile'.\", None)\n",                                                                                                                                                                                       
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

Reply:
Sorry for the delayed response.

The error was due to filename, it should be of format `<datastore>/<filename>`

I have updated the same in the documentation, examples and integration test.